lib/Elastica/Util.php

```diff
@@ -11,6 +11,73 @@
  */
 class Util
 {
+    /** @var array */
+    protected static $dateMathSymbols = ['<', '>', '/', '{', '}', '|', '+', ':', ','];
+
+    /** @var array */
+    protected static $escapedDateMathSymbols = ['%3C', '%3E', '%2F', '%7B', '%7D', '%7C', '%2B', '%3A', '%2C'];
+
+    /**
+     * Checks if date math is already escaped within request URI.
+     *
+     * @param string $requestUri
+     *
+     * @return bool
+     */
+    public static function isDateMathEscaped($requestUri)
+    {
+        // In practice, the only symbol that really needs to be escaped in URI is '/' => '%2F'
+        return false !== strpos($requestUri, '%2F');
+    }
+
+    /**
+     * Escapes date math symbols within request URI.
+     * @see https://www.elastic.co/guide/en/elasticsearch/reference/5.x/date-math-index-names.html
+     *
+     * @param string $requestUri
+     *
+     * @return string
+     */
+    public static function escapeDateMath($requestUri)
+    {
+        // No need to escape if a second slash is not present
+        if (empty($requestUri) || false === strpos($requestUri, '/', 1)) {
+            return $requestUri;
+        }
+
+        // Check if date math if used at all. Find last '>'. E.g. /<log-{now/d}>,log-2011.12.01/log/_refresh
+        $pos1 = strrpos($requestUri, '>');
+        if (false === $pos1) {
+            return $requestUri;
+        }
+
+        // Trim leading slash if present
+        if ($startsWithSlash = '/' === $requestUri[0]) {
```

Comment:
Not sure if we should keep this check, as Http and HttpAdapter assume that `/` is present, but Guzzle transport trims it and reappends. Opinions?